src/sources/docker_logs.rs

```diff
@@ -543,26 +515,11 @@ impl DockerLogsSource {
         }
     }
 
-    /// True if container with the given id and image must be excluded from logging,
-    /// because it's a vector instance, probably this one.
-    fn exclude_vector<'a>(&self, id: &str, image: impl Into<Option<&'a str>>) -> bool {
-        if self.exclude_self {
-            let hostname_hint = self
-                .hostname
-                .as_ref()
-                .map(|maybe_short_id| id.starts_with(maybe_short_id))
-                .unwrap_or(false);
-            let image_hint = image
-                .into()
-                .map(|image| image.starts_with(VECTOR_IMAGE_NAME))
-                .unwrap_or(false);
-            if hostname_hint || image_hint {
-                // This container is probably itself.
-                info!(message = "Detected self container.", id = %id);
-                return false;
-            }
-        }
-        true
+    fn exclude_self<'a>(&self, id: &str) -> bool {
```

Comment:
Just calling attention to the failing check here. It seems like you could drop the lifetime.

Reply:
Thanks! Removed.